Lib/test/test_hashlib.py

```diff
@@ -1185,6 +1185,15 @@ def test_file_digest(self):
             with open(os_helper.TESTFN, "wb") as f:
                 hashlib.file_digest(f, "sha256")
 
+        class NonBlocking:
+            def readinto(self, buf):
+                return None
+            def readable(self):
+                return True
+
+        with self.assertRaises(BlockingIOError):
```

Comment:
This test is good for checking that the non-blocking error is raised, but doesn't validate that the hash changes in this case. Is it possible to add a test that the hash changes?